bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -457,4 +491,82 @@ public static boolean isValidIPConfig(String ipAddress) {
         return false;
     }
 
+    private void scheduleToPollNetworkInterface(Map<String, Object> props) {
+
+        final NetUtil myself = this;
+        if (networkInterfacePollFuture != null) {
+            networkInterfacePollFuture.cancel(true);
+            networkInterfacePollFuture = null;
+        }
+
+        Runnable tempRunnable = new Runnable() {
+            @Override
+            public void run() {
+                myself.pollAndNotifyNetworkInterfaceAddress();
+            }
+        };
+
+        String pollFrequenceSecondsStr = "";
+        int pollFrequenceSeconds = INTERFACE_POLL_DEFAULT_FREQUENCY_SECONDS;
+        try {
+            pollFrequenceSecondsStr = (String) props.get(INTERFACE_POLL_FREQUENCY_SECONDS_KEY);
+            pollFrequenceSeconds = Integer.parseInt(pollFrequenceSecondsStr);
+        } catch (NumberFormatException e) {
+            LOGGER.debug("Cannot parse value {} from key {}, will use default {}", pollFrequenceSecondsStr,
+                    INTERFACE_POLL_FREQUENCY_SECONDS_KEY, INTERFACE_POLL_DEFAULT_FREQUENCY_SECONDS);
+        }
+        networkInterfacePollFuture = (ScheduledFuture<Void>) scheduledExecutorService
+                .scheduleWithFixedDelay(tempRunnable, 1, INTERFACE_POLL_DEFAULT_FREQUENCY_SECONDS, TimeUnit.SECONDS);
+    }
+
+    private void pollAndNotifyNetworkInterfaceAddress() {
+        Collection<CidrAddress> newInterfaceAddresses = getAllInterfaceAddresses();
+        if (networkAddressChangeListeners.isEmpty()) {
+            // no listeners listening, just update
+            lastKnownInterfaceAddresses = newInterfaceAddresses;
+            return;
+        }
+
+        // Look for added addresses to notify
+        List<CidrAddress> added = newInterfaceAddresses.stream()
+                .filter(newInterfaceAddr -> !lastKnownInterfaceAddresses.contains(newInterfaceAddr))
+                .collect(Collectors.toList());
+
+        // Look for removed addresses to notify
+        List<CidrAddress> removed = lastKnownInterfaceAddresses.stream()
+                .filter(lastKnownInterfaceAddr -> !newInterfaceAddresses.contains(lastKnownInterfaceAddr))
+                .collect(Collectors.toList());
+
+        lastKnownInterfaceAddresses = newInterfaceAddresses;
+
+        if (!added.isEmpty() || !removed.isEmpty()) {
+            notifyListeners(added, removed);
+        }
+    }
+
+    private void notifyListeners(List<CidrAddress> added, List<CidrAddress> removed) {
+        // Prevent listeners changing the list
+        List<CidrAddress> unmodifiableAddedList = Collections.unmodifiableList(added);
+        List<CidrAddress> unmodifiableRemovedList = Collections.unmodifiableList(removed);
+        networkAddressChangeListeners
+                .forEach(listener -> listener.onChanged(unmodifiableAddedList, unmodifiableRemovedList));
```

Comment:
Listeners might throw exceptions - you should make sure that one "bad" listener cannot break the whole notification mechanism, so that others aren't notified anymore. Also, do you have any contract wrt return time expectation? In any case, you should use our SafeCaller infrastructure for doing such calls in external code.

Reply:
Great suggestion